### ISSUE TITLE
Managed routes should not apply to destination zt member

### ISSUE THREAD

**Issue by jdrews** [zerotier/ZeroTierOne#352]
Giving 1.1.8 a try with managed routes. Ran into an issue with managed routes being applied to the actual gateway member, and then that member is unable to route into other networks.

Lets look at my example from #323 

> ZeroTier network IPv4 10.1.1.0/24
> 
> Member 0000001 10.1.1.1 office-router
> 
> Member 0000023 10.1.1.23 road-warrior23
> push-route 172.25.3.0/24 10.1.1.1
> push-route 192.168.62.0/24 10.1.1.1

Where 10.1.1.1 is a ZeroTier member with access into 172.25.3.0/24 and 192.168.62.0/24 and capable of routing 10.1.1.0/24 traffic into said networks.

So in my.zerotier.com I set managed routes to

> 172.25.3.0/24 10.1.1.1
> 192.168.62.0/24 10.1.1.1

Now if you look at 10.1.1.1 you see the following:
```
root@office-router:~# netstat -rn
Kernel IP routing table
Destination     Gateway         Genmask         Flags   MSS Window  irtt Iface
0.0.0.0         172.25.1.1        0.0.0.0         UG        0 0          0 eth0
172.25.3.0        10.1.1.1       255.255.255.128 UG        0 0          0 zt0
172.25.3.128      10.1.1.1       255.255.255.128 UG        0 0          0 zt0
192.168.62.0        10.1.1.1       255.255.255.128 UG        0 0          0 zt0
192.168.62.128      10.1.1.1       255.255.255.128 UG        0 0          0 zt0
```

So 10.1.1.1 is unable to actually route packets it receives into the networks it's connected to. I suppose the fix would be not apply routes to a member if the member owns the destination IP. 

office-router (10.1.1.1) is debian 8 with zt 1.1.8

With ZeroTier off this is what netstat says 
```
root@office-router:~# netstat -rn
Kernel IP routing table
Destination     Gateway         Genmask         Flags   MSS Window  irtt Iface
0.0.0.0         172.25.1.1        0.0.0.0         UG        0 0          0 eth0
172.25.3.0        172.25.1.1        255.255.255.0   UG        0 0          0 eth0
192.168.62.0        172.25.1.1        255.255.255.0   UG        0 0          0 eth0
```

**Comment by adamierymenko** [comment]
I believe you're correct -- this should not behave this way. For some reason it did not cause a problem in our testing, but then again we did more tests with default gateway override than with this kind of scenario.

The correct behavior should be to not add routes 'via' the current node. This is simple to implement.

We might push another version this week to address this and another issue related to IPv6 default routes.

**Comment by adamierymenko** [comment]
@jdrews If you want you can now pull the `dev` branch, type `make` and `make install` on your router (other hosts should not need to be updated) and try again.

The router should now avoid setting routes via itself. This check is performed against any IP assigned to the router's ZeroTier tap interface whether or not it is a managed IP. So 10.1.1.1 should not set any routes via 10.1.1.1 regardless of the value of the `allowManaged` setting.

Let me know if this fixes the issue. If so a new release is coming very soon with this little fix in it.

**Comment by adamierymenko** [comment]
I believe this issue is fixed in 1.1.10 -- at least it seems to be in our testing. Closing for now but feel free to comment or re-open if you still see route problems.

Thanks for testing!

**Comment by adamierymenko** [comment]
Fix is fixed because fix was broken. Pushing another release. 😦

**Comment by jdrews** [comment]
Confirmed fixed in 1.1.12. Thanks!

**Comment by adamierymenko** [comment]
Check this out as well... similar to what you're doing:

https://www.zerotier.com/community/topic/72/creating-a-zerotier-to-amazon-vpc-gateway
